docs/aws/audit/iamcompliance/rules/iam_groups_policy_blacklisted_check_remediation.mdx

```diff
@@ -1,6 +1,302 @@
+2. **Initialize Boto3 Client**: Set up the IAM client using Boto3.
+3. **Define the Inline Policy**: Create a JSON structure for the inline policy to block specific KMS actions.
+4. **Attach the Inline Policy**: Use the `put_user_policy` method to attach the policy to an IAM user.
+
+By following these steps, you can prevent blocked KMS actions in inline policies using Python scripts.
+</Accordion>
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console and open the IAM console at https://console.aws.amazon.com/iam/.
+
+2. In the navigation pane, choose "Policies". This will open a list of all the IAM policies that are currently configured in your AWS environment.
+
+3. Select the policy you want to check for blocked KMS actions. This will open the policy details page.
+
+4. In the policy details page, check the policy document for any "Deny" statements that are applied to KMS actions. If there are any "Deny" statements applied to KMS actions, then the policy is blocking those actions.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure you have the necessary permissions to execute IAM related commands.
+
+2. Once the AWS CLI is set up, you can list all the IAM policies using the following command:
+   ```
+   aws iam list-policies --scope Local
+   ```
+   This command will return a list of all the IAM policies that are created within your AWS account.
+
+3. For each policy, you can get the policy details including the policy document by using the following command:
+   ```
+   aws iam get-policy-version --policy-arn <Policy_ARN> --version-id <Policy_Version_ID>
+   ```
+   Replace `<Policy_ARN>` with the ARN of the policy and `<Policy_Version_ID>` with the version ID of the policy. This command will return the policy document which includes the permissions set by the policy.
+
+4. Now, you need to check the policy document for any blocked KMS actions. You can do this by looking for "kms:Deny" statements in the policy document. If you find any such statements, it means that some KMS actions are blocked in the inline policy. You can use a JSON parser or a script to automate this process. For example, in Python, you can use the `json` module to parse the policy document and check for blocked KMS actions.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3):
+   You need to install and configure Boto3 to interact with AWS services. You can install it using pip:
+   ```
+   pip install boto3
+   ```
+   Then, configure your AWS credentials either by setting the following environment variables:
+   ```
+   AWS_ACCESS_KEY_ID = 'your_access_key'
+   AWS_SECRET_ACCESS_KEY = 'your_secret_key'
```

Comment:
<h2>Secret Detected: Secret Keyword</h2> <a href="https://console.cloudanix.com/code?t=secrets"> Click here for more details</a>